src/core/viewports/viewport.js

```diff
@@ -132,26 +132,27 @@ export default class Viewport {
       this.meterOffset = modelMatrix ? modelMatrix.transformVector(position) : position;
     }
 
-    // Determine camera center
-    this.center = this.isGeospatial ?
-      getMercatorWorldPosition({
-        longitude, latitude, zoom: this.zoom, meterOffset: this.meterOffset
-      }) :
-      position;
-
-    // console.log(this.scale, this.distanceScales.pixelsPerMeter);
-
     this.viewMatrixUncentered = viewMatrix;
 
-    // Make a centered version of the matrix for projection modes without an offset
-    this.viewMatrix = new Matrix4()
+    if (this.isGeospatial) {
+      // Determine camera center
+      this.center = getMercatorWorldPosition({
+        longitude, latitude, zoom: this.zoom, meterOffset: this.meterOffset
+      });
+
+      // Make a centered version of the matrix for projection modes without an offset
+      this.viewMatrix = new Matrix4()
       // Apply the uncentered view matrix
-      .multiplyRight(this.viewMatrixUncentered)
-      // The Mercator world coordinate system is upper left,
-      // but GL expects lower left, so we flip it around the center after all transforms are done
-      .scale([1, -1, 1])
-      // And center it
-      .translate(new Vector3(this.center || ZERO_VECTOR).negate());
+        .multiplyRight(this.viewMatrixUncentered)
+        // The Mercator world coordinate system is upper left,
+        // but GL expects lower left, so we flip it around the center after all transforms are done
+        .scale([1, -1, 1])
+        // And center it
+        .translate(new Vector3(this.center || ZERO_VECTOR).negate());
+    } else {
```

Comment:
> .translate(new Vector3(this.center || ZERO_VECTOR).negate());

Should we do this for non geospatial cases too?

Reply:
shouldn't do this in non-geospatial cases, cause every centering should be handled view matrix already.

Reply:
I would like us to handle `position` in non-geospatial viewports as well, that would allow us to truly unify around the new `viewState` class. Does setting this.center below actually have any effect?

For the geospatial case it is handled by the meterOffset in the center calculation.

Reply:
Also to avoid confusion should we rename `position` to 'centerOffset`, because that what it is used as to calculate meterOffset.

Reply:
I don't think we have a test case that uses 'position' in non geo-spatial viewport, we can come back to this in a different PR.

Reply:
> Also to avoid confusion should we rename `position` to 'centerOffset`, because that what it is used as to calculate meterOffset.

The idea is have been pursuing is that `Viewports` are designed to show a `ViewState`. A `ViewState` is fundamentally two things:
*  a `position` (which defaults to `[0, 0, 0]`)
*  a `direction` (which can be expressed with `pitch` and `bearing` or a vector etc). 
* an optional geospatial anchor (`lng`, `lat`) that is only used in geospatial modes.

So `position` is the single most important aspect of the `ViewState` which is the main parameter to the `ViewPort`.

So, if one agrees with this view, it would not make sense to rename `position` to `centerOffset`.